### PR TITLE
custom_auth_in_javascript flag and setAuthTokenListener RN.Android implementation

### PR DESCRIPTION
**Pull request by dhei**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

1. Introduce `custom_auth_in_javascript` flag in App Center config file to avoid token history wipe.

2. Add RN.Android `setAuthTokenListener` implementation.

3. TestApp UI changes will be in another PR.

## Related PRs or issues

[AB#67957](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/67957)
[AB#67955](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/67955)

### CODE REVIEW

File: TestApp/android/app/src/main/assets/appcenter-config.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
     "app_secret": "32fcfc69-d576-41dc-8d49-4be159e3d7b2",
-    "enable_push_in_javascript": true
+    "enable_push_in_javascript": true,
+    "custom_auth_in_javascript": true
```
Comment (guperrot) [inline TestApp/android/app/src/main/assets/appcenter-config.json]: was that name validated with PM?

File: appcenter/AppCenter.d.ts [hunk 2]
```diff
@@ -37,3 +37,9 @@ export function setEnabled(enabled: boolean): Promise<void>;
 export function setUserId(userId: string | null): Promise<void>;
 export function getInstallId(): Promise<string>;
 export function setCustomProperties(properties: CustomProperties);
+export function setAuthToken(authToken: string | null): void;
+export function setAuthTokenListener(listener: AuthTokenListener) : void;
+
+export interface AuthTokenListener {
+    acquireAuthToken?: () => string;
```
Comment (guperrot) [inline appcenter/AppCenter.d.ts]: is the callback optional? or did you mean you can return string | null

Comment (guperrot) [reply appcenter/AppCenter.d.ts]: or both maybe

File: appcenter/android/src/main/java/com/microsoft/appcenter/reactnative/appcenter/AppCenterReactNativeListener.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+public class AppCenterReactNativeListener implements AuthTokenListener {
+
+    private static final String ON_ACQUIRE_AUTH_TOKEN_EVENT = "OnAcquireAuthTokenEvent";
+
+    private ReactApplicationContext mReactApplicationContext;
+
+    private AuthTokenCallback mCallback;
+
+    private List<String> mPendingEvents = new ArrayList<>();
```
Comment (guperrot) [inline appcenter/android/src/main/java/com/microsoft/appcenter/reactnative/appcenter/AppCenterReactNativeListener.java]: There should be 0 or 1

File: appcenter/AppCenter.d.ts (hunk 2 above)
Comment (dhei) [reply appcenter/AppCenter.d.ts]: Good catch. Fixed now.

**Comment by dhei** [comment]
Closing this PR and replacing with 2 smaller PRs.

**Pull request closed without merging**